module/DTO/questionsDTO.js

```diff
@@ -0,0 +1,577 @@
+            return reject({
+              success: false,
+              message: res.params.errmsg,
+              status: httpStatusCode.bad_request.status,
+            });
+          }
+
+          // Extract the question set hierarchy from the response
+          const questionSetHierarchy =
+            res?.result?.questionSet || res?.result?.questionset;
+
+          // Gather questions from the hierarchy based on the criteria provided
+          const migratedCriteriaQuestions =
+            questionSetHierarchy?.children || [];
+          const evidences = await this.questionEvidences(
+            migratedCriteriaQuestions,
+            submissionDocumentCriterias,
+            isPageQuestionsRequired
+          )?.data;
+
+          // Format the assessment template with relevant evidence data
+          assessmentTemplate.assessment.evidences[0].name = capitalize(
+            solutionDocument?.type
+          );
+          assessmentTemplate.assessment.evidences[0].sections[0].name = `${capitalize(
+            solutionDocument?.type
+          )} Questions`;
+          assessmentTemplate.assessment.evidences[0].sections[0].questions =
+            evidences?.evidenceSections || [];
+          assessmentTemplate.assessment.evidences[0].sections[0].code = "SQ";
+          assessmentTemplate.assessment.evidences[0].code =
+            assessmentTemplate.assessment.evidences[0].externalId = "SF";
+          assessmentTemplate.assessment.evidences[0].description =
+            questionSetHierarchy?.description || "";
+
+          // Cache the formatted evidence data
+          await setKey(
+            solutionDocument.referenceQuestionSetId,
+            {
+              ...evidences,
+              evidences: assessmentTemplate.assessment.evidences,
+            },
+            cacheTtl
+          );
+
+          // Resolve with the formatted evidence data
+          resolve({
+            success: true,
+            message: messageConstants.apiResponses.EVIDENCE_FETCHED,
+            data: {
+              ...evidences,
+              evidences: assessmentTemplate.assessment.evidences,
+            },
+          });
+        }
+      } catch (error) {
+        // Handle any errors that occur during the process
+        return reject({
+          success: false,
+          message: error.message,
+          data: false,
+        });
+      }
+    });
+  }
+
+  /**
+   * Processes criteria questions and generates evidence sections.
+   * @method
+   * @name questionEvidences
+   * @param {Array} criteriaQuestions - Array of criteria questions.
+   * @param {Array} submissionDocumentCriterias - Array of submission document criteria objects to update.
+   * @param {Boolean} isPageQuestionsRequired - Flag to determine if page questions are required for each criteria.
+   * @returns {Promise<Object>} - Resolves with an object containing evidence sections and updated submission document criteria.
```

Comment:
@jayesh9747 please provide sample input data and response of the function everywhere, then we can understand later, Do this for every function in the code you have written